File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeyssubmission/model/ClientTemporaryExposureKeysPayload.java
package uk.nhs.nhsx.diagnosiskeyssubmission.model;

import com.fasterxml.jackson.annotation.JsonCreator;

import java.util.List;
import java.util.UUID;

import static java.util.Objects.requireNonNull;

public class ClientTemporaryExposureKeysPayload {

    public final UUID diagnosisKeySubmissionToken;
    public final List<ClientTemporaryExposureKey> temporaryExposureKeys;

    @JsonCreator
    public ClientTemporaryExposureKeysPayload(UUID diagnosisKeySubmissionToken,
                                              List<ClientTemporaryExposureKey> temporaryExposureKeys) {
        this.diagnosisKeySubmissionToken = requireNonNull(diagnosisKeySubmissionToken);
        this.temporaryExposureKeys = requireNonNull(temporaryExposureKeys);
    }
}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/diagnosiskeyssubmission/model/ClientTemporaryExposureKeysPayload.kt
package uk.nhs.nhsx.diagnosiskeyssubmission.model

import uk.nhs.nhsx.domain.TestKit
import java.util.*

data class ClientTemporaryExposureKeysPayload(
    val diagnosisKeySubmissionToken: UUID? = UUID.fromString("00000000-0000-0000-0000-000000000000"),
    val temporaryExposureKeys: List<ClientTemporaryExposureKey?>,
    val isPrivateJourney: Boolean? = false,
    val testKit: TestKit? = TestKit.LAB_RESULT
)
